// vue-banana-i18n/dist/vue-banana-i18n.mjs
import { inject as Fu, reactive as Gu } from "vue";
var Pu = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function Su(l) {
  return l && l.__esModule && Object.prototype.hasOwnProperty.call(l, "default") ? l.default : l;
}
var tu = { exports: {} };
(function(l, p) {
  (function(f, g) {
    l.exports = g();
  })(Pu, function() {
    var f = { ar: "٠١٢٣٤٥٦٧٨٩", fa: "۰۱۲۳۴۵۶۷۸۹", ml: "൦൧൨൩൪൫൬൭൮൯", kn: "೦೧೨೩೪೫೬೭೮೯", lo: "໐໑໒໓໔໕໖໗໘໙", or: "୦୧୨୩୪୫୬୭୮୯", kh: "០១២៣៤៥៦៧៨៩", nqo: "߀߁߂߃߄߅߆߇߈߉", pa: "੦੧੨੩੪੫੬੭੮੯", gu: "૦૧૨૩૪૫૬૭૮૯", hi: "०१२३४५६७८९", my: "၀၁၂၃၄၅၆၇၈၉", ta: "௦௧௨௩௪௫௬௭௮௯", te: "౦౧౨౩౪౫౬౭౮౯", th: "๐๑๒๓๔๕๖๗๘๙", bo: "༠༡༢༣༤༥༦༧༨༩" }, g = { ab: ["ru"], abs: ["id"], ace: ["id"], ady: ["ady-cyrl"], aeb: ["aeb-arab"], "aeb-arab": ["ar"], aln: ["sq"], alt: ["ru"], ami: ["zh-hant"], an: ["es"], anp: ["hi"], arn: ["es"], arq: ["ar"], ary: ["ar"], arz: ["ar"], ast: ["es"], atj: ["fr"], av: ["ru"], avk: ["fr", "es", "ru"], awa: ["hi"], ay: ["es"], azb: ["fa"], ba: ["ru"], ban: ["id"], "ban-bali": ["ban"], bar: ["de"], bbc: ["bbc-latn"], "bbc-latn": ["id"], bcc: ["fa"], "be-tarask": ["be"], bgn: ["fa"], bh: ["bho"], bi: ["en"], bjn: ["id"], bm: ["fr"], bpy: ["bn"], bqi: ["fa"], br: ["fr"], btm: ["id"], bug: ["id"], bxr: ["ru"], ca: ["oc"], "cbk-zam": ["es"], cdo: ["nan", "zh-hant"], ce: ["ru"], co: ["it"], crh: ["crh-latn"], "crh-cyrl": ["ru"], cs: ["sk"], csb: ["pl"], cv: ["ru"], "de-at": ["de"], "de-ch": ["de"], "de-formal": ["de"], dsb: ["hsb", "de"], dtp: ["ms"], dty: ["ne"], egl: ["it"], eml: ["it"], "en-ca": ["en"], "en-gb": ["en"], "es-419": ["es"], "es-formal": ["es"], ext: ["es"], ff: ["fr"], fit: ["fi"], frc: ["fr"], frp: ["fr"], frr: ["de"], fur: ["it"], gag: ["tr"], gan: ["gan-hant", "zh-hant", "zh-hans"], "gan-hans": ["zh-hans"], "gan-hant": ["zh-hant", "zh-hans"], gcr: ["fr"], gl: ["pt"], glk: ["fa"], gn: ["es"], gom: ["gom-deva"], "gom-deva": ["hi"], gor: ["id"], gsw: ["de"], guc: ["es"], hak: ["zh-hant"], hif: ["hif-latn"], hrx: ["de"], hsb: ["dsb", "de"], ht: ["fr"], "hu-formal": ["hu"], hyw: ["hy"], ii: ["zh-cn", "zh-hans"], inh: ["ru"], io: ["eo"], iu: ["ike-cans"], jam: ["en"], jut: ["da"], jv: ["id"], kaa: ["kk-latn", "kk-cyrl"], kab: ["fr"], kbd: ["kbd-cyrl"], kbp: ["fr"], khw: ["ur"], kiu: ["tr"], kjp: ["my"], kk: ["kk-cyrl"], "kk-arab": ["kk-cyrl"], "kk-cn": ["kk-arab", "kk-cyrl"], "kk-kz": ["kk-cyrl"], "kk-latn": ["kk-cyrl"], "kk-tr": ["kk-latn", "kk-cyrl"], kl: ["da"], "ko-kp": ["ko"], koi: ["ru"], krc: ["ru"], krl: ["fi"], ks: ["ks-arab"], ksh: ["de"], ku: ["ku-latn"], "ku-arab": ["ckb"], kum: ["ru"], kv: ["ru"], lad: ["es"], lb: ["de"], lbe: ["ru"], lez: ["ru", "az"], li: ["nl"], lij: ["it"], liv: ["et"], lki: ["fa"], lld: ["it", "rm", "fur"], lmo: ["pms", "eml", "lij", "vec", "it"], ln: ["fr"], lrc: ["fa"], ltg: ["lv"], luz: ["fa"], lzh: ["zh-hant"], lzz: ["tr"], mad: ["id"], mai: ["hi"], "map-bms": ["jv", "id"], mdf: ["myv", "ru"], mg: ["fr"], mhr: ["mrj", "ru"], min: ["id"], mnw: ["my"], mo: ["ro"], mrj: ["mhr", "ru"], "ms-arab": ["ms"], mwl: ["pt"], myv: ["mdf", "ru"], mzn: ["fa"], nah: ["es"], nan: ["cdo", "zh-hant"], nap: ["it"], nb: ["nn"], nds: ["de"], "nds-nl": ["nl"], nia: ["id"], "nl-informal": ["nl"], nn: ["nb"], nrm: ["fr"], oc: ["ca", "fr"], olo: ["fi"], os: ["ru"], pcd: ["fr"], pdc: ["de"], pdt: ["de"], pfl: ["de"], pih: ["en"], pms: ["it"], pnt: ["el"], pt: ["pt-br"], "pt-br": ["pt"], qu: ["qug", "es"], qug: ["qu", "es"], rgn: ["it"], rmy: ["ro"], "roa-tara": ["it"], rue: ["uk", "ru"], rup: ["ro"], ruq: ["ruq-latn", "ro"], "ruq-cyrl": ["mk"], "ruq-latn": ["ro"], sa: ["hi"], sah: ["ru"], scn: ["it"], sco: ["en"], sdc: ["it"], sdh: ["cbk", "fa"], ses: ["fr"], sg: ["fr"], sgs: ["lt"], sh: ["bs", "sr-el", "hr"], shi: ["fr"], shy: ["shy-latn"], "shy-latn": ["fr"], sk: ["cs"], skr: ["skr-arab"], "skr-arab": ["ur", "pnb"], sli: ["de"], smn: ["fi"], sr: ["sr-ec"], srn: ["nl"], stq: ["de"], sty: ["ru"], su: ["id"], szl: ["pl"], szy: ["zh-tw", "zh-hant", "zh-hans"], tay: ["zh-tw", "zh-hant", "zh-hans"], tcy: ["kn"], tet: ["pt"], tg: ["tg-cyrl"], trv: ["zh-tw", "zh-hant", "zh-hans"], tt: ["tt-cyrl", "ru"], "tt-cyrl": ["ru"], ty: ["fr"], tyv: ["ru"], udm: ["ru"], ug: ["ug-arab"], vec: ["it"], vep: ["et"], vls: ["nl"], vmf: ["de"], vot: ["fi"], vro: ["et"], wa: ["fr"], wo: ["fr"], wuu: ["zh-hans"], xal: ["ru"], xmf: ["ka"], yi: ["he"], za: ["zh-hans"], zea: ["nl"], zgh: ["kab"], zh: ["zh-hans"], "zh-cn": ["zh-hans"], "zh-hant": ["zh-hans"], "zh-hk": ["zh-hant", "zh-hans"], "zh-mo": ["zh-hk", "zh-hant", "zh-hans"], "zh-my": ["zh-sg", "zh-hans"], "zh-sg": ["zh-hans"], "zh-tw": ["zh-hant", "zh-hans"] };
    class i {
      constructor(e) {
        this.locale = e;
      }
      convertPlural(e, d) {
        const r = /\d+=/i;
        if (!d || d.length === 0)
          return "";
        for (let a = 0; a < d.length; a++) {
          const c = d[a];
          if (r.test(c)) {
            if (parseInt(c.slice(0, c.indexOf("=")), 10) === e)
              return c.slice(c.indexOf("=") + 1);
            d[a] = void 0;
          }
        }
        d = d.filter((a) => !!a);
        let n = this.getPluralForm(e, this.locale);
        return n = Math.min(n, d.length - 1), d[n];
      }
      getPluralForm(e, d) {
        const r = new Intl.PluralRules(d), n = r.resolvedOptions().pluralCategories, a = r.select(e);
        return ["zero", "one", "two", "few", "many", "other"].filter((c) => n.includes(c)).indexOf(a);
      }
      convertNumber(e, d = !1) {
        let r = this.digitTransformTable(this.locale), n = "";
        if (d) {
          if (parseFloat(e, 10) === e || !r)
            return e;
          const a = [];
          for (const o in r)
            a[r[o]] = o;
          r = a;
          const c = String(e);
          for (let o = 0; o < c.length; o++)
            n += r[c[o]] || c[o];
          return parseFloat(n, 10);
        }
        if (Intl.NumberFormat) {
          let a;
          const c = [...g[this.locale] || [], "en"];
          return a = Intl.NumberFormat.supportedLocalesOf(this.locale).length ? [this.locale] : c, n = new Intl.NumberFormat(a).format(e), n === "NaN" && (n = e), n;
        }
      }
      convertGrammar(e, d) {
        return e;
      }
      gender(e, d) {
        if (!d || d.length === 0)
          return "";
        for (; d.length < 2; )
          d.push(d[d.length - 1]);
        return e === "male" ? d[0] : e === "female" ? d[1] : d.length === 3 ? d[2] : d[0];
      }
      digitTransformTable(e) {
        return !!f[e] && f[e].split("");
      }
    }
    var z = { bs: class extends i {
      convertGrammar(u, e) {
        switch (e) {
          case "instrumental":
            u = "s " + u;
            break;
          case "lokativ":
            u = "o " + u;
        }
        return u;
      }
    }, default: i, dsb: class extends i {
      convertGrammar(u, e) {
        switch (e) {
          case "instrumental":
            u = "z " + u;
            break;
          case "lokatiw":
            u = "wo " + u;
        }
        return u;
      }
    }, fi: class extends i {
      convertGrammar(u, e) {
        let d = u.match(/[aou][^äöy]*$/i);
        const r = u;
        switch (u.match(/wiki$/i) && (d = !1), u.match(/[bcdfghjklmnpqrstvwxz]$/i) && (u += "i"), e) {
          case "genitive":
            u += "n";
            break;
          case "elative":
            u += d ? "sta" : "stä";
            break;
          case "partitive":
            u += d ? "a" : "ä";
            break;
          case "illative":
            u += u.slice(-1) + "n";
            break;
          case "inessive":
            u += d ? "ssa" : "ssä";
            break;
          default:
            u = r;
        }
        return u;
      }
    }, ga: class extends i {
      convertGrammar(u, e) {
        if (e === "ainmlae")
          switch (u) {
            case "an Domhnach":
              u = "Dé Domhnaigh";
              break;
            case "an Luan":
              u = "Dé Luain";
              break;
            case "an Mháirt":
              u = "Dé Mháirt";
              break;
            case "an Chéadaoin":
              u = "Dé Chéadaoin";
              break;
            case "an Déardaoin":
              u = "Déardaoin";
              break;
            case "an Aoine":
              u = "Dé hAoine";
              break;
            case "an Satharn":
              u = "Dé Sathairn";
          }
        return u;
      }
    }, he: class extends i {
      convertGrammar(u, e) {
        switch (e) {
          case "prefixed":
          case "תחילית":
            u.slice(0, 1) === "ו" && u.slice(0, 2) !== "וו" && (u = "ו" + u), u.slice(0, 1) === "ה" && (u = u.slice(1)), (u.slice(0, 1) < "א" || u.slice(0, 1) > "ת") && (u = "־" + u);
        }
        return u;
      }
    }, hsb: class extends i {
      convertGrammar(u, e) {
        switch (e) {
          case "instrumental":
            u = "z " + u;
            break;
          case "lokatiw":
            u = "wo " + u;
        }
        return u;
      }
    }, hu: class extends i {
      convertGrammar(u, e) {
        switch (e) {
          case "rol":
            u += "ról";
            break;
          case "ba":
            u += "ba";
            break;
          case "k":
            u += "k";
        }
        return u;
      }
    }, hy: class extends i {
      convertGrammar(u, e) {
        return e === "genitive" && (u.slice(-1) === "ա" ? u = u.slice(0, -1) + "այի" : u.slice(-1) === "ո" ? u = u.slice(0, -1) + "ոյի" : u.slice(-4) === "գիրք" ? u = u.slice(0, -4) + "գրքի" : u += "ի"), u;
      }
    }, la: class extends i {
      convertGrammar(u, e) {
        switch (e) {
          case "genitive":
            u = (u = (u = (u = (u = (u = (u = (u = (u = u.replace(/u[ms]$/i, "i")).replace(/ommunia$/i, "ommunium")).replace(/a$/i, "ae")).replace(/libri$/i, "librorum")).replace(/nuntii$/i, "nuntiorum")).replace(/tio$/i, "tionis")).replace(/ns$/i, "ntis")).replace(/as$/i, "atis")).replace(/es$/i, "ei");
            break;
          case "accusative":
            u = (u = (u = (u = (u = (u = (u = (u = (u = u.replace(/u[ms]$/i, "um")).replace(/ommunia$/i, "am")).replace(/a$/i, "ommunia")).replace(/libri$/i, "libros")).replace(/nuntii$/i, "nuntios")).replace(/tio$/i, "tionem")).replace(/ns$/i, "ntem")).replace(/as$/i, "atem")).replace(/es$/i, "em");
            break;
          case "ablative":
            u = (u = (u = (u = (u = (u = (u = (u = (u = u.replace(/u[ms]$/i, "o")).replace(/ommunia$/i, "ommunibus")).replace(/a$/i, "a")).replace(/libri$/i, "libris")).replace(/nuntii$/i, "nuntiis")).replace(/tio$/i, "tione")).replace(/ns$/i, "nte")).replace(/as$/i, "ate")).replace(/es$/i, "e");
        }
        return u;
      }
    }, os: class extends i {
      convertGrammar(u, e) {
        let d, r, n, a;
        switch (d = "мæ", r = "", n = "", a = "", u.match(/тæ$/i) ? (u = u.slice(0, -1), d = "æм") : u.match(/[аæеёиоыэюя]$/i) ? r = "й" : u.match(/у$/i) ? u.slice(-2, -1).match(/[аæеёиоыэюя]$/i) || (r = "й") : u.match(/[бвгджзйклмнопрстфхцчшщьъ]$/i) || (n = "-"), e) {
          case "genitive":
            a = n + r + "ы";
            break;
          case "dative":
            a = n + r + "æн";
            break;
          case "allative":
            a = n + d;
            break;
          case "ablative":
            a = r === "й" ? n + r + "æ" : n + r + "æй";
            break;
          case "superessive":
            a = n + r + "ыл";
            break;
          case "equative":
            a = n + r + "ау";
            break;
          case "comitative":
            a = n + "имæ";
        }
        return u + a;
      }
    }, ru: class extends i {
      convertGrammar(u, e) {
        return e === "genitive" && (u.slice(-1) === "ь" ? u = u.slice(0, -1) + "я" : u.slice(-2) === "ия" ? u = u.slice(0, -2) + "ии" : u.slice(-2) === "ка" ? u = u.slice(0, -2) + "ки" : u.slice(-2) === "ти" ? u = u.slice(0, -2) + "тей" : u.slice(-2) === "ды" ? u = u.slice(0, -2) + "дов" : u.slice(-3) === "ник" && (u = u.slice(0, -3) + "ника")), u;
      }
    }, sl: class extends i {
      convertGrammar(u, e) {
        switch (e) {
          case "mestnik":
            u = "o " + u;
            break;
          case "orodnik":
            u = "z " + u;
        }
        return u;
      }
    }, uk: class extends i {
      convertGrammar(u, e) {
        switch (e) {
          case "genitive":
            u.slice(-1) === "ь" ? u = u.slice(0, -1) + "я" : u.slice(-2) === "ія" ? u = u.slice(0, -2) + "ії" : u.slice(-2) === "ка" ? u = u.slice(0, -2) + "ки" : u.slice(-2) === "ти" ? u = u.slice(0, -2) + "тей" : u.slice(-2) === "ды" ? u = u.slice(0, -2) + "дов" : u.slice(-3) === "ник" && (u = u.slice(0, -3) + "ника");
            break;
          case "accusative":
            u.slice(-2) === "ія" && (u = u.slice(0, -2) + "ію");
        }
        return u;
      }
    } };
    const x = new RegExp("(?:([A-Za-zªµºÀ-ÖØ-öø-ʸʻ-ˁːˑˠ-ˤˮͰ-ͳͶͷͺ-ͽͿΆΈ-ΊΌΎ-ΡΣ-ϵϷ-҂Ҋ-ԯԱ-Ֆՙ-՟ա-և։ः-हऻऽ-ीॉ-ौॎ-ॐक़-ॡ।-ঀংঃঅ-ঌএঐও-নপ-রলশ-হঽ-ীেৈোৌৎৗড়ঢ়য়-ৡ০-ৱ৴-৺ਃਅ-ਊਏਐਓ-ਨਪ-ਰਲਲ਼ਵਸ਼ਸਹਾ-ੀਖ਼-ੜਫ਼੦-੯ੲ-ੴઃઅ-ઍએ-ઑઓ-નપ-રલળવ-હઽ-ીૉોૌૐૠૡ૦-૰ૹଂଃଅ-ଌଏଐଓ-ନପ-ରଲଳଵ-ହଽାୀେୈୋୌୗଡ଼ଢ଼ୟ-ୡ୦-୷ஃஅ-ஊஎ-ஐஒ-கஙசஜஞடணதந-பம-ஹாிுூெ-ைொ-ௌௐௗ௦-௲ఁ-ఃఅ-ఌఎ-ఐఒ-నప-హఽు-ౄౘ-ౚౠౡ౦-౯౿ಂಃಅ-ಌಎ-ಐಒ-ನಪ-ಳವ-ಹಽ-ೄೆ-ೈೊೋೕೖೞೠೡ೦-೯ೱೲംഃഅ-ഌഎ-ഐഒ-ഺഽ-ീെ-ൈൊ-ൌൎൗൟ-ൡ൦-൵൹-ൿංඃඅ-ඖක-නඳ-රලව-ෆා-ෑෘ-ෟ෦-෯ෲ-෴ก-ะาำเ-ๆ๏-๛ກຂຄງຈຊຍດ-ທນ-ຟມ-ຣລວສຫອ-ະາຳຽເ-ໄໆ໐-໙ໜ-ໟༀ-༗༚-༴༶༸༾-ཇཉ-ཬཿ྅ྈ-ྌ྾-࿅࿇-࿌࿎-࿚က-ာေးျြဿ-ၗၚ-ၝၡ-ၰၵ-ႁႃႄႇ-ႌႎ-ႜ႞-ჅჇჍა-ቈቊ-ቍቐ-ቖቘቚ-ቝበ-ኈኊ-ኍነ-ኰኲ-ኵኸ-ኾዀዂ-ዅወ-ዖዘ-ጐጒ-ጕጘ-ፚ፠-፼ᎀ-ᎏᎠ-Ᏽᏸ-ᏽᐁ-ᙿᚁ-ᚚᚠ-ᛸᜀ-ᜌᜎ-ᜑᜠ-ᜱ᜵᜶ᝀ-ᝑᝠ-ᝬᝮ-ᝰក-ឳាើ-ៅះៈ។-៚ៜ០-៩᠐-᠙ᠠ-ᡷᢀ-ᢨᢪᢰ-ᣵᤀ-ᤞᤣ-ᤦᤩ-ᤫᤰᤱᤳ-ᤸ᥆-ᥭᥰ-ᥴᦀ-ᦫᦰ-ᧉ᧐-᧚ᨀ-ᨖᨙᨚ᨞-ᩕᩗᩡᩣᩤᩭ-ᩲ᪀-᪉᪐-᪙᪠-᪭ᬄ-ᬳᬵᬻᬽ-ᭁᭃ-ᭋ᭐-᭪᭴-᭼ᮂ-ᮡᮦᮧ᮪ᮮ-ᯥᯧᯪ-ᯬᯮ᯲᯳᯼-ᰫᰴᰵ᰻-᱉ᱍ-᱿᳀-᳇᳓᳡ᳩ-ᳬᳮ-ᳳᳵᳶᴀ-ᶿḀ-ἕἘ-Ἕἠ-ὅὈ-Ὅὐ-ὗὙὛὝὟ-ώᾀ-ᾴᾶ-ᾼιῂ-ῄῆ-ῌῐ-ΐῖ-Ίῠ-Ῥῲ-ῴῶ-ῼ‎ⁱⁿₐ-ₜℂℇℊ-ℓℕℙ-ℝℤΩℨK-ℭℯ-ℹℼ-ℿⅅ-ⅉⅎ⅏Ⅰ-ↈ⌶-⍺⎕⒜-ⓩ⚬⠀-⣿Ⰰ-Ⱞⰰ-ⱞⱠ-ⳤⳫ-ⳮⳲⳳⴀ-ⴥⴧⴭⴰ-ⵧⵯ⵰ⶀ-ⶖⶠ-ⶦⶨ-ⶮⶰ-ⶶⶸ-ⶾⷀ-ⷆⷈ-ⷎⷐ-ⷖⷘ-ⷞ々-〇〡-〩〮〯〱-〵〸-〼ぁ-ゖゝ-ゟァ-ヺー-ヿㄅ-ㄭㄱ-ㆎ㆐-ㆺㇰ-㈜㈠-㉏㉠-㉻㉿-㊰㋀-㋋㋐-㋾㌀-㍶㍻-㏝㏠-㏾㐀-䶵一-鿕ꀀ-ꒌꓐ-ꘌꘐ-ꘫꙀ-ꙮꚀ-ꚝꚠ-ꛯ꛲-꛷Ꜣ-ꞇ꞉-ꞭꞰ-ꞷꟷ-ꠁꠃ-ꠅꠇ-ꠊꠌ-ꠤꠧ꠰-꠷ꡀ-ꡳꢀ-ꣃ꣎-꣙ꣲ-ꣽ꤀-ꤥ꤮-ꥆꥒ꥓꥟-ꥼꦃ-ꦲꦴꦵꦺꦻꦽ-꧍ꧏ-꧙꧞-ꧤꧦ-ꧾꨀ-ꨨꨯꨰꨳꨴꩀ-ꩂꩄ-ꩋꩍ꩐-꩙꩜-ꩻꩽ-ꪯꪱꪵꪶꪹ-ꪽꫀꫂꫛ-ꫫꫮ-ꫵꬁ-ꬆꬉ-ꬎꬑ-ꬖꬠ-ꬦꬨ-ꬮꬰ-ꭥꭰ-ꯤꯦꯧꯩ-꯬꯰-꯹가-힣ힰ-ퟆퟋ-ퟻ-舘並-龎ﬀ-ﬆﬓ-ﬗＡ-Ｚａ-ｚｦ-ﾾￂ-ￇￊ-ￏￒ-ￗￚ-ￜ]|\uD800[\uDC00-\uDC0B]|\uD800[\uDC0D-\uDC26]|\uD800[\uDC28-\uDC3A]|𐀼|𐀽|\uD800[\uDC3F-\uDC4D]|\uD800[\uDC50-\uDC5D]|\uD800[\uDC80-\uDCFA]|𐄀|𐄂|\uD800[\uDD07-\uDD33]|\uD800[\uDD37-\uDD3F]|\uD800[\uDDD0-\uDDFC]|\uD800[\uDE80-\uDE9C]|\uD800[\uDEA0-\uDED0]|\uD800[\uDF00-\uDF23]|\uD800[\uDF30-\uDF4A]|\uD800[\uDF50-\uDF75]|\uD800[\uDF80-\uDF9D]|\uD800[\uDF9F-\uDFC3]|\uD800[\uDFC8-\uDFD5]|\uD801[\uDC00-\uDC9D]|\uD801[\uDCA0-\uDCA9]|\uD801[\uDD00-\uDD27]|\uD801[\uDD30-\uDD63]|𐕯|\uD801[\uDE00-\uDF36]|\uD801[\uDF40-\uDF55]|\uD801[\uDF60-\uDF67]|𑀀|\uD804[\uDC02-\uDC37]|\uD804[\uDC47-\uDC4D]|\uD804[\uDC66-\uDC6F]|\uD804[\uDC82-\uDCB2]|𑂷|𑂸|\uD804[\uDCBB-\uDCC1]|\uD804[\uDCD0-\uDCE8]|\uD804[\uDCF0-\uDCF9]|\uD804[\uDD03-\uDD26]|𑄬|\uD804[\uDD36-\uDD43]|\uD804[\uDD50-\uDD72]|\uD804[\uDD74-\uDD76]|\uD804[\uDD82-\uDDB5]|\uD804[\uDDBF-\uDDC9]|𑇍|\uD804[\uDDD0-\uDDDF]|\uD804[\uDDE1-\uDDF4]|\uD804[\uDE00-\uDE11]|\uD804[\uDE13-\uDE2E]|𑈲|𑈳|𑈵|\uD804[\uDE38-\uDE3D]|\uD804[\uDE80-\uDE86]|𑊈|\uD804[\uDE8A-\uDE8D]|\uD804[\uDE8F-\uDE9D]|\uD804[\uDE9F-\uDEA9]|\uD804[\uDEB0-\uDEDE]|\uD804[\uDEE0-\uDEE2]|\uD804[\uDEF0-\uDEF9]|𑌂|𑌃|\uD804[\uDF05-\uDF0C]|𑌏|𑌐|\uD804[\uDF13-\uDF28]|\uD804[\uDF2A-\uDF30]|𑌲|𑌳|\uD804[\uDF35-\uDF39]|\uD804[\uDF3D-\uDF3F]|\uD804[\uDF41-\uDF44]|𑍇|𑍈|\uD804[\uDF4B-\uDF4D]|𑍐|𑍗|\uD804[\uDF5D-\uDF63]|\uD805[\uDC80-\uDCB2]|𑒹|\uD805[\uDCBB-\uDCBE]|𑓁|\uD805[\uDCC4-\uDCC7]|\uD805[\uDCD0-\uDCD9]|\uD805[\uDD80-\uDDB1]|\uD805[\uDDB8-\uDDBB]|𑖾|\uD805[\uDDC1-\uDDDB]|\uD805[\uDE00-\uDE32]|𑘻|𑘼|𑘾|\uD805[\uDE41-\uDE44]|\uD805[\uDE50-\uDE59]|\uD805[\uDE80-\uDEAA]|𑚬|𑚮|𑚯|𑚶|\uD805[\uDEC0-\uDEC9]|\uD805[\uDF00-\uDF19]|𑜠|𑜡|𑜦|\uD805[\uDF30-\uDF3F]|\uD806[\uDCA0-\uDCF2]|𑣿|\uD806[\uDEC0-\uDEF8]|\uD808[\uDC00-\uDF99]|\uD809[\uDC00-\uDC6E]|\uD809[\uDC70-\uDC74]|\uD809[\uDC80-\uDD43]|\uD80C[\uDC00-\uDFFF]|\uD80D[\uDC00-\uDC2E]|\uD811[\uDC00-\uDE46]|\uD81A[\uDC00-\uDE38]|\uD81A[\uDE40-\uDE5E]|\uD81A[\uDE60-\uDE69]|𖩮|𖩯|\uD81A[\uDED0-\uDEED]|𖫵|\uD81A[\uDF00-\uDF2F]|\uD81A[\uDF37-\uDF45]|\uD81A[\uDF50-\uDF59]|\uD81A[\uDF5B-\uDF61]|\uD81A[\uDF63-\uDF77]|\uD81A[\uDF7D-\uDF8F]|\uD81B[\uDF00-\uDF44]|\uD81B[\uDF50-\uDF7E]|\uD81B[\uDF93-\uDF9F]|𛀀|𛀁|\uD82F[\uDC00-\uDC6A]|\uD82F[\uDC70-\uDC7C]|\uD82F[\uDC80-\uDC88]|\uD82F[\uDC90-\uDC99]|𛲜|𛲟|\uD834[\uDC00-\uDCF5]|\uD834[\uDD00-\uDD26]|\uD834[\uDD29-\uDD66]|\uD834[\uDD6A-\uDD72]|𝆃|𝆄|\uD834[\uDD8C-\uDDA9]|\uD834[\uDDAE-\uDDE8]|\uD834[\uDF60-\uDF71]|\uD835[\uDC00-\uDC54]|\uD835[\uDC56-\uDC9C]|𝒞|𝒟|𝒢|𝒥|𝒦|\uD835[\uDCA9-\uDCAC]|\uD835[\uDCAE-\uDCB9]|𝒻|\uD835[\uDCBD-\uDCC3]|\uD835[\uDCC5-\uDD05]|\uD835[\uDD07-\uDD0A]|\uD835[\uDD0D-\uDD14]|\uD835[\uDD16-\uDD1C]|\uD835[\uDD1E-\uDD39]|\uD835[\uDD3B-\uDD3E]|\uD835[\uDD40-\uDD44]|𝕆|\uD835[\uDD4A-\uDD50]|\uD835[\uDD52-\uDEA5]|\uD835[\uDEA8-\uDEDA]|\uD835[\uDEDC-\uDF14]|\uD835[\uDF16-\uDF4E]|\uD835[\uDF50-\uDF88]|\uD835[\uDF8A-\uDFC2]|\uD835[\uDFC4-\uDFCB]|\uD836[\uDC00-\uDDFF]|\uD836[\uDE37-\uDE3A]|\uD836[\uDE6D-\uDE74]|\uD836[\uDE76-\uDE83]|\uD836[\uDE85-\uDE8B]|\uD83C[\uDD10-\uDD2E]|\uD83C[\uDD30-\uDD69]|\uD83C[\uDD70-\uDD9A]|\uD83C[\uDDE6-\uDE02]|\uD83C[\uDE10-\uDE3A]|\uD83C[\uDE40-\uDE48]|🉐|🉑|[\uD840-\uD868][\uDC00-\uDFFF]|\uD869[\uDC00-\uDED6]|\uD869[\uDF00-\uDFFF]|[\uD86A-\uD86C][\uDC00-\uDFFF]|\uD86D[\uDC00-\uDF34]|\uD86D[\uDF40-\uDFFF]|\uD86E[\uDC00-\uDC1D]|\uD86E[\uDC20-\uDFFF]|[\uD86F-\uD872][\uDC00-\uDFFF]|\uD873[\uDC00-\uDEA1]|\uD87E[\uDC00-\uDE1D]|[\uDB80-\uDBBE][\uDC00-\uDFFF]|\uDBBF[\uDC00-\uDFFD]|[\uDBC0-\uDBFE][\uDC00-\uDFFF]|\uDBFF[\uDC00-\uDFFD])|([֐־׀׃׆׈-׿߀-ߪߴߵߺ-ࠕࠚࠤࠨ࠮-ࡘ࡜-࢟‏יִײַ-ﬨשׁ-ﭏ؈؋؍؛-ي٭-ٯٱ-ەۥۦۮۯۺ-ܐܒ-ܯ݋-ޥޱ-޿ࢠ-࣢ﭐ-ﴽ﵀-﷏ﷰ-﷼﷾﷿ﹰ-﻾]|\uD802[\uDC00-\uDD1E]|\uD802[\uDD20-\uDE00]|𐨄|\uD802[\uDE07-\uDE0B]|\uD802[\uDE10-\uDE37]|\uD802[\uDE3B-\uDE3E]|\uD802[\uDE40-\uDEE4]|\uD802[\uDEE7-\uDF38]|\uD802[\uDF40-\uDFFF]|\uD803[\uDC00-\uDE5F]|\uD803[\uDE7F-\uDFFF]|\uD83A[\uDC00-\uDCCF]|\uD83A[\uDCD7-\uDFFF]|\uD83B[\uDC00-\uDDFF]|\uD83B[\uDF00-\uDFFF]|\uD83B[\uDF00-\uDFFF]|\uD83B[\uDF00-\uDFFF]|\uD83B[\uDF00-\uDFFF]|\uD83B[\uDF00-\uDFFF]|\uD83B[\uDF00-\uDFFF]|\uD83B[\uDF00-\uDFFF]|\uD83B[\uDF00-\uDFFF]|\uD83B[\uDF00-\uDFFF]|\uD83B[\uDF00-\uDFFF]|\uD83B[\uDF00-\uDFFF]|\uD83B[\uDF00-\uDFFF]|\uD83B[\uDF00-\uDFFF]|\uD83B[\uDE00-\uDEEF]|\uD83B[\uDEF2-\uDEFF]))");
    class D {
      constructor(e) {
        this.locale = e, this.language = new (z[e] || z.default)(e);
      }
      emit(e, d) {
        let r, n, a;
        switch (typeof e) {
          case "string":
          case "number":
            r = e;
            break;
          case "object":
            if (n = e.slice(1).map((c) => this.emit(c, d)), a = e[0].toLowerCase(), typeof this[a] != "function")
              throw new Error('unknown operation "' + a + '"');
            r = this[a](n, d);
            break;
          case "undefined":
            r = "";
            break;
          default:
            throw new Error("unexpected type in AST: " + typeof e);
        }
        return r;
      }
      concat(e) {
        let d = "";
        return e.forEach((r) => {
          d += r;
        }), d;
      }
      replace(e, d) {
        const r = parseInt(e[0], 10);
        return r < d.length ? d[r] : "$" + (r + 1);
      }
      plural(e) {
        const d = parseFloat(this.language.convertNumber(e[0], 10)), r = e.slice(1);
        return r.length ? this.language.convertPlural(d, r) : "";
      }
      gender(e) {
        const d = e[0], r = e.slice(1);
        return this.language.gender(d, r);
      }
      grammar(e) {
        const d = e[0], r = e[1];
        return r && d && this.language.convertGrammar(r, d);
      }
      wikilink(e) {
        let d, r = e[0];
        r.charAt(0) === ":" && (r = r.slice(1));
        const n = `./${r}`;
        return d = e.length === 1 ? r : e[1], `<a href="${n}" title="${r}">${d}</a>`;
      }
      extlink(e) {
        if (e.length !== 2)
          throw new Error("Expected two items in the node");
        return `<a href="${e[0]}">${e[1]}</a>`;
      }
      bidi(e) {
        const d = function(r) {
          const n = r.match(x);
          return n ? n[2] === void 0 ? "ltr" : "rtl" : null;
        }(e[0]);
        return d === "ltr" ? "‪" + e[0] + "‬" : d === "rtl" ? "‫" + e[0] + "‬" : e[0];
      }
      formatnum(e) {
        const d = !!e[1] && e[1] === "R", r = e[0];
        return typeof r == "string" || typeof r == "number" ? this.language.convertNumber(r, d) : r;
      }
      htmlattributes(e) {
        const d = {};
        for (let r = 0, n = e.length; r < n; r += 2)
          d[e[r]] = e[r + 1];
        return d;
      }
      htmlelement(e) {
        const d = e.shift(), r = e.shift();
        let n = e, a = "";
        for (const c in r)
          a += ` ${c}="${r[c]}"`;
        return Array.isArray(n) || (n = [n]), `<${d}${a}>${n.join("")}</${d}>`;
      }
    }
    class H {
      constructor(e, { wikilinks: d = !1 } = {}) {
        this.locale = e, this.wikilinks = d, this.emitter = new D(this.locale);
      }
      parse(e, d) {
        if (e.includes("{{") || e.includes("<") || this.wikilinks && e.includes("[")) {
          const r = function(n, { wikilinks: a = !1 } = {}) {
            let c = 0;
            function o(t) {
              return () => {
                for (let s = 0; s < t.length; s++) {
                  const k = t[s]();
                  if (k !== null)
                    return k;
                }
                return null;
              };
            }
            function h(t) {
              const s = c, k = [];
              for (let v = 0; v < t.length; v++) {
                const y = t[v]();
                if (y === null)
                  return c = s, null;
                k.push(y);
              }
              return k;
            }
            function m(t, s) {
              return () => {
                const k = c, v = [];
                let y = s();
                for (; y !== null; )
                  v.push(y), y = s();
                return v.length < t ? (c = k, null) : v;
              };
            }
            function b(t) {
              const s = t.length;
              return () => {
                let k = null;
                return n.slice(c, c + s) === t && (k = t, c += s), k;
              };
            }
            function w(t) {
              return () => {
                const s = n.slice(c).match(t);
                return s === null ? null : (c += s[0].length, s[0]);
              };
            }
            const R = w(/^\s+/), B = b("|"), N = b(":"), au = b("\\"), su = w(/^./), cu = b("$"), lu = w(/^\d+/), Z = b('"'), K = b("'"), iu = w(a ? /^[^{}[\]$<\\]/ : /^[^{}$<\\]/), ou = w(a ? /^[^{}[\]$\\|]/ : /^[^{}$\\|]/), fu = o([O, w(a ? /^[^{}[\]$\s]/ : /^[^{}$\s]/)]);
            function O() {
              const t = h([au, su]);
              return t === null ? null : t[1];
            }
            const hu = o([O, ou]), mu = o([O, iu]);
            function A() {
              const t = h([cu, lu]);
              return t === null ? null : ["REPLACE", parseInt(t[1], 10) - 1];
            }
            const C = (_ = w(/^[ !"$&'()*,./0-9;=?@A-Z^_`a-z~\x80-\xFF+-]+/), U = function(t) {
              return t.toString();
            }, () => {
              const t = _();
              return t === null ? null : U(t);
            });
            var _, U;
            function W() {
              const t = h([B, m(0, E)]);
              if (t === null)
                return null;
              const s = t[1];
              return s.length > 1 ? ["CONCAT"].concat(s) : s[0];
            }
            function bu() {
              const t = h([C, N, A]);
              return t === null ? null : [t[0], t[2]];
            }
            function ku() {
              const t = h([C, N, E]);
              return t === null ? null : [t[0], t[2]];
            }
            function pu() {
              const t = h([C, N]);
              return t === null ? null : [t[0], ""];
            }
            const gu = o([function() {
              const t = h([o([bu, ku, pu]), m(0, W)]);
              return t === null ? null : t[0].concat(t[1]);
            }, function() {
              const t = h([C, m(0, W)]);
              return t === null ? null : [t[0]].concat(t[1]);
            }]), wu = b("{{"), vu = b("}}"), yu = b("[["), zu = b("]]"), $u = b("["), xu = b("]");
            function I() {
              const t = h([wu, gu, vu]);
              return t === null ? null : t[1];
            }
            const Au = o([function() {
              const t = h([m(1, E), B, m(1, j)]);
              return t === null ? null : [["CONCAT"].concat(t[0]), ["CONCAT"].concat(t[2])];
            }, function() {
              const t = h([m(1, E)]);
              return t === null ? null : [["CONCAT"].concat(t[0])];
            }]);
            function X() {
              let t = null;
              const s = h([yu, Au, zu]);
              if (s !== null) {
                const k = s[1];
                t = ["WIKILINK"].concat(k);
              }
              return t;
            }
            function J() {
              let t = null;
              const s = h([$u, m(1, Mu), R, m(1, j), xu]);
              return s !== null && (t = ["EXTLINK", s[1].length === 1 ? s[1][0] : ["CONCAT"].concat(s[1]), ["CONCAT"].concat(s[3])]), t;
            }
            const L = w(/^[A-Za-z]+/);
            function Cu() {
              const t = w(/^[^"]*/), s = h([Z, t, Z]);
              return s === null ? null : s[1];
            }
            function ju() {
              const t = w(/^[^']*/), s = h([K, t, K]);
              return s === null ? null : s[1];
            }
            function Eu() {
              const t = w(/^\s*=\s*/), s = h([R, L, t, o([Cu, ju])]);
              return s === null ? null : [s[1], s[3]];
            }
            function Tu() {
              const t = m(0, Eu)();
              return Array.prototype.concat.apply(["HTMLATTRIBUTES"], t);
            }
            const Mu = o([I, A, X, J, function() {
              const t = m(1, fu)();
              return t === null ? null : t.join("");
            }]), j = o([I, A, X, J, function() {
              let t = null;
              const s = c, k = b("<"), v = w(/^\/?/), y = w(/^\s*>/), q = h([k, L, Tu, v, y]);
              if (q === null)
                return null;
              const Q = c, Y = q[1], F = m(0, j)(), Nu = c, uu = h([b("</"), L, y]);
              if (uu === null)
                return ["CONCAT", n.slice(s, Q)].concat(F);
              const Ou = c, Iu = uu[1], eu = q[2];
              if (function($, T, G, P = { allowedHtmlElements: ["b", "bdi", "del", "i", "ins", "u", "font", "big", "small", "sub", "sup", "h1", "h2", "h3", "h4", "h5", "h6", "cite", "code", "em", "s", "strike", "strong", "tt", "var", "div", "center", "blockquote", "ol", "ul", "dl", "table", "caption", "pre", "ruby", "rb", "rp", "rt", "rtc", "p", "span", "abbr", "dfn", "kbd", "samp", "data", "time", "mark", "li", "dt", "dd"], allowedHtmlCommonAttributes: ["id", "class", "style", "lang", "dir", "title", "aria-describedby", "aria-flowto", "aria-hidden", "aria-label", "aria-labelledby", "aria-owns", "role", "about", "property", "resource", "datatype", "typeof", "itemid", "itemprop", "itemref", "itemscope", "itemtype"], allowedHtmlAttributesByElement: {} }) {
                if (($ = $.toLowerCase()) !== (T = T.toLowerCase()) || P.allowedHtmlElements.indexOf($) === -1)
                  return !1;
                const Lu = /[\000-\010\013\016-\037\177]|expression|filter\s*:|accelerator\s*:|-o-link\s*:|-o-link-source\s*:|-o-replace\s*:|url\s*\(|image\s*\(|image-set\s*\(/i;
                for (let M = 0, qu = G.length; M < qu; M += 2) {
                  const S = G[M];
                  if (P.allowedHtmlCommonAttributes.indexOf(S) === -1 && (P.allowedHtmlAttributesByElement[$] || []).indexOf(S) === -1 || S === "style" && G[M + 1].search(Lu) !== -1)
                    return !1;
                }
                return !0;
              }(Y, Iu, eu.slice(1)))
                t = ["HTMLELEMENT", Y, eu].concat(F);
              else {
                const $ = (T) => T.replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;").replace(/"/g, "&quot;").replace(/'/g, "&#039;");
                t = ["CONCAT", $(n.slice(s, Q))].concat(F, $(n.slice(Nu, Ou)));
              }
              return t;
            }, function() {
              const t = m(1, mu)();
              return t === null ? null : t.join("");
            }]), E = o([I, A, function() {
              const t = m(1, hu)();
              return t === null ? null : t.join("");
            }]), V = function() {
              const t = m(0, j)();
              return t === null ? null : ["CONCAT"].concat(t);
            }();
            if (V === null || c !== n.length)
              throw new Error("Parse error at position " + c.toString() + " in input: " + n);
            return V;
          }(e, { wikilinks: this.wikilinks });
          return this.emitter.emit(r, d);
        }
        return this.simpleParse(e, d);
      }
      simpleParse(e, d) {
        return e.replace(/\$(\d+)/g, (r, n) => {
          const a = parseInt(n, 10) - 1;
          return d[a] !== void 0 ? d[a] : "$" + n;
        });
      }
    }
    class nu {
      constructor(e) {
        this.sourceMap = /* @__PURE__ */ new Map();
      }
      load(e, d) {
        if (typeof e != "object")
          throw new Error("Invalid message source. Must be an object");
        if (d) {
          if (!/^[a-zA-Z0-9-]+$/.test(d))
            throw new Error(`Invalid locale ${d}`);
          for (const r in e)
            if (r.indexOf("@") !== 0) {
              if (typeof e[r] == "object")
                return this.load(e);
              if (typeof e[r] != "string")
                throw new Error(`Invalid message for message ${r} in ${d} locale.`);
              break;
            }
          this.sourceMap.has(d) ? this.sourceMap.set(d, Object.assign(this.sourceMap.get(d), e)) : this.sourceMap.set(d, e);
        } else
          for (d in e)
            this.load(e[d], d);
      }
      getMessage(e, d) {
        const r = this.sourceMap.get(d);
        return r ? r[e] : null;
      }
      hasLocale(e) {
        return this.sourceMap.has(e);
      }
    }
    return class {
      constructor(u, { finalFallback: e = "en", messages: d, wikilinks: r = !1 } = {}) {
        this.locale = u, this.parser = new H(this.locale, { wikilinks: r }), this.messageStore = new nu(), d && this.load(d, this.locale), this.finalFallback = e, this.wikilinks = r;
      }
      load(u, e) {
        return this.messageStore.load(u, e || this.locale);
      }
      i18n(u, ...e) {
        return this.parser.parse(this.getMessage(u), e);
      }
      setLocale(u) {
        this.locale = u, this.parser = new H(this.locale, { wikilinks: this.wikilinks });
      }
      getFallbackLocales() {
        return [...g[this.locale] || [], this.finalFallback];
      }
      getMessage(u) {
        let e = this.locale, d = 0;
        const r = this.getFallbackLocales(this.locale);
        for (; e; ) {
          const n = e.split("-");
          let a = n.length;
          do {
            const c = n.slice(0, a).join("-"), o = this.messageStore.getMessage(u, c);
            if (typeof o == "string")
              return o;
            a--;
          } while (a);
          e = r[d], d++;
        }
        return u;
      }
      registerParserPlugin(u, e) {
        D.prototype[u] = e;
      }
    };
  });
})(tu);
var Du = tu.exports;
const Hu = /* @__PURE__ */ Su(Du), du = (l) => {
  let p, f;
  if (Array.isArray(l.value) ? (p = l.arg, f = l.value) : l.value !== null && typeof l.value == "object" ? (p = l.value.msg, f = l.value.params) : p = l.arg || l.value, f = f || [], Array.isArray(f) || (f = [f]), !p)
    throw new Error(
      `${l.rawName} used with parameter array but without message key`
    );
  return { msg: p, params: f };
}, ru = Symbol("banana-context");
function Bu() {
  const l = Fu(ru);
  if (!l)
    throw new Error("No i18n provided!!!");
  return l;
}
function Zu(l = { messages: {}, locale: "en", wikilinks: !0 }) {
  const p = Gu(new Hu(l.locale, l));
  return {
    install: (f) => {
      f.provide(ru, p), f.config.globalProperties.$i18n = (g, ...i) => p.i18n(g, ...i), f.provide("setLocale", (g) => {
        p.setLocale(g);
      }), f.directive("i18n", (g, i) => {
        const { msg: z, params: x } = du(i);
        i.modifiers.html ? g.innerHTML = p.i18n(z, ...x) : g.textContent = p.i18n(z, ...x);
      }), f.directive("i18n-html", (g, i) => {
        const { msg: z, params: x } = du(i);
        g.innerHTML = p.i18n(z, ...x);
      });
    }
  };
}
export {
  Zu as createI18n,
  Bu as useI18n
};
